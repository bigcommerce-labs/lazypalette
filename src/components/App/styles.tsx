import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
