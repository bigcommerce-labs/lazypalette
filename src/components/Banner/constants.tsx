import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
