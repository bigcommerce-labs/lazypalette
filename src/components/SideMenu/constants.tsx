export const enum Tips {
    Primary = 'Collapse menu',
    Secondary = 'Expand menu',
}

export const ThemeStatus = {
    active: {
        label: 'ACTIVE THEME',
        tooltip: 'This theme is currently applied to your store.',
    },
    inactive: {
        label: 'INACTIVE THEME',
        tooltip: 'This theme is not currently applied to your store.',
    },
    preview: {
        label: 'THEME PREVIEW',
        tooltip: 'To use this theme, you must buy it before you can save or publish to your store.',
    },
};

export const enum Links {
    Help = 'https://support.bigcommerce.com/articles/Public/Store-Design',
}
