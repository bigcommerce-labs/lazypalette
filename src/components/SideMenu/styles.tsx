import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

export const activeClassName = 'nav-item-active';

export const BackLink = styled.div`
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.larger};
    margin-bottom: 1.75rem;
`;

export const BackLinkIcon = styled.span`
    float: left;
    position: relative;
    top: 0.25rem;
`;

interface NavProps {
    activeClassName: string;
}

export const NavItem = styled(NavLink)
    .attrs<NavProps>({
    activeClassName,
})`
    ${({ theme }) => `
        color: ${theme.colors.primaryText};
        display: block;
        height: 2.25rem;
        overflow: hidden;
        text-decoration: none;

        &:visited {
            color: ${theme.colors.primaryText};
        }

        &.${activeClassName} {
            font-weight: ${theme.typography.fontWeight.bold};
        }
    `}
`;

export const StyledSideMenu = styled.nav`
    height: 100%;
    padding-top: 1.5rem;
    padding-left: 2rem;
    padding-right: 2rem;
    width: 9.5rem;
`;

export const StyledMenuItems = styled.ul`
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    line-height: 2.25rem;
    list-style-type: none;
    margin-top: 0;
    padding-left: 0;
    vertical-align: top;
`;

export const StyledMenuItemIcon = styled.span`
    float: right;
    position: relative;
    top: 0.25rem;
`;

export const DesignMenuButtons = styled.div`
    display: flex;
    justify-content: space-around;
`;
