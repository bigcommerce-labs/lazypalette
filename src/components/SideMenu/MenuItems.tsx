import { Icon } from 'pattern-lab';
import React, { Component } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';

import ThemeSettings from '../ThemeSettings/ThemeSettings';
import ThemeVariations from '../ThemeVariations/ThemeVariations';

import { activeClassName, NavItem, StyledMenuItems, StyledMenuItemIcon } from './styles';

interface MenuItem {
    disabled?: boolean;
    label: string;
    path: string;
}

interface MenuItemsProps extends RouteComponentProps<{}> {
    items: MenuItem[];
    showArrows?: boolean;
    currentPath: string;
}

const ExpandMenuRoutes = ({ route }: { route: string }) => {
    if (route === 'theme') {
        return <ThemeVariations/>;
    }

    if (route.startsWith('style/')) {
        return <ThemeSettings settingsIndex={parseInt(route.split('/')[1], 10)}/>;
    }

    return null;
};

class MenuItems extends Component<MenuItemsProps, {}> {
    static defaultProps = {
        showArrows: false,
    };

    isPathActive = (path: string) => (
        this.props.location.pathname === `${this.props.currentPath}${path}`
    );

    toggleLink = (path: string) => (
        this.isPathActive(path) ? this.props.currentPath : `${this.props.currentPath}${path}`
    );

    render() {
        return (
            <StyledMenuItems>
                {this.props.items.map(({ disabled, path, label }) => (
                    <li key={path}>
                        <ExpandMenuRoutes route={path}/>
                        <NavItem
                            to={this.toggleLink(path)}
                            exact
                            isActive={(match, location) => this.isPathActive(path)}
                            activeClassName={activeClassName}
                            disabled={disabled}>
                            {label}
                            {this.props.showArrows &&
                                <StyledMenuItemIcon>
                                    <Icon glyph="chevronRight" size="large" />
                                </StyledMenuItemIcon>
                            }
                        </NavItem>
                    </li>
                ))}
            </StyledMenuItems>
        );
    }
}

export default withRouter(MenuItems);
