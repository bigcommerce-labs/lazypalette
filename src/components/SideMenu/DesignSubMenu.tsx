import React, { PureComponent } from 'react';

import { appRoutes } from '../Routes/Routes';

import MenuItems from './MenuItems';

interface MenuItem {
    divider?: boolean;
    label: string;
    path: string;
    externalLink?: boolean;
}

interface DesignSubMenuProps {
    currentPath: string;
    isPreview: boolean;
    sections: string[];
}

const staticTopItems: MenuItem[] = [
    {
        label: 'Styles',
        path: appRoutes.styles.route,
    },
];

const staticBottomPreviewItems: MenuItem[] = [
    {
        divider: true,
        externalLink: true,
        label: 'Help',
        path: 'https://support.bigcommerce.com/articles/Public/Stencil-Themes/#enabling',
    },
];

const staticBottomItems: MenuItem[] = [
    {
        divider: true,
        label: 'History',
        path: appRoutes.history.route,
    },
    {
        label: 'More Options',
        path: appRoutes.options.route,
    },
    {
        externalLink: true,
        label: 'Help',
        path: 'https://support.bigcommerce.com/articles/Public/Stencil-Themes/#enabling',
    },
];

const getItems = (sections: string[]) => {
    return sections.map((section: string, index: number) => (
        {
            label: section,
            path: `${appRoutes.section.route}${index}`,
        }
    ));
};

class DesignSubMenu extends PureComponent<DesignSubMenuProps> {
    render() {
        const { currentPath, isPreview } = this.props;
        const items = [...staticTopItems, ...getItems(this.props.sections)];

        if (isPreview) {
            items.push(...staticBottomPreviewItems);
        } else {
            items.push(...staticBottomItems);
        }

        return (
            <MenuItems
                items={items}
                currentPath={currentPath}
                showArrows={true}
            />
        );
    }
}

export default DesignSubMenu;
