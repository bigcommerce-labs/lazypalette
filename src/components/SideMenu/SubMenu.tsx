import React, { Component } from 'react';
import styled from 'styled-components';

import { activeClassName, NavItem } from './styles';
import MenuItems from './MenuItems';

const BackLink = styled.div`
  color: #34343B;
  font-size: 24px;
  margin-bottom: 30px;
  margin-left: 28px;
`;

interface SubMenuItem {
  label: string;
  path: string;
}

interface SubMenuProps {
  title: string;
  items?: SubMenuItem[];
  match: any;
}

class SubMenu extends Component<SubMenuProps, any> {
  static defaultProps: Partial<SubMenuProps> = {
    items: [],
  };

  constructor(props: SubMenuProps) {
    super(props);
  }

  render() {
    return (
      <div>
        <BackLink>
          <NavItem to="/" exact activeClassName={activeClassName}>&lt; {this.props.title}</NavItem>
        </BackLink>
        <MenuItems items={this.props.items!} match={this.props.match}/>
      </div>
    );
  }
}

export default SubMenu;
