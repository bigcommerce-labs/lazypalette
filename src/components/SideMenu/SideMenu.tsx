import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Route, RouteComponentProps } from 'react-router-dom';

import { State } from '../../reducers/reducers';

import { StyledSideMenu } from './styles';
import DesignSubMenu from './DesignSubMenu';
import MenuItems from './MenuItems';
import SubMenu from './SubMenu';

interface SideMenuProps extends RouteComponentProps<{}> {
    themeDesignSections: string[];
}

const items = [
    {
        label: 'Design',
        path: 'design',
    },
    {
        label: 'Pages and Layout',
        path: 'pages',
    },
    {
        label: 'Apps',
        path: 'apps',
    },
    {
        label: 'History',
        path: 'history',
    },
];

const routes = [
    {
        path: 'pages',
        submenu_title: 'Pages',
    },
    {
        path: 'apps',
        submenu_title: 'Apps',
    },
    {
        path: 'history',
        submenu_title: 'History',
    },
];

class SideMenu extends Component<SideMenuProps, {}> {
    render() {
        return (
            <StyledSideMenu>
                <Route
                    path="/"
                    exact
                    render={({ match }) => <MenuItems
                        items={items.map(({ label, path }) => ({ label, path }))}
                        currentPath={match.path}
                    />}
                />
                <Route
                    path="/design/"
                    render={({ match }) => <DesignSubMenu sections={this.props.themeDesignSections}
                        currentPath={match.path}/>}
                />
                {routes.map(route => (
                    <Route
                        key={route.path}
                        path={`/${route.path}/`}
                        render={({ match }) => (
                            <SubMenu title={route.submenu_title} currentPath={match.path}/>)}
                    />
                ))}
            </StyledSideMenu>
        );
    }
}

const mapStateToProps = (state: State) => {
    return {
        themeDesignSections: state.theme.schema.map(({ name }) => name),
    };
};

export default withRouter(connect(mapStateToProps)(SideMenu));
