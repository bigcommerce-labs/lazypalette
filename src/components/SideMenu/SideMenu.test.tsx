import { shallow } from 'enzyme';
import React from 'react';
import { StaticRouter } from 'react-router';

import { Collapsed } from './constants';
import { CollapseButton } from './styles';
import { SideMenu } from './SideMenu';

describe('SideMenu', () => {
    const mockpostThemeConfig = jest.fn();
    const mockSettings = {
        'alert-color': '#fff',
        'body-font': 'Google_Karla_400',
        brandpage_products_per_page: 12,
        show_accept_amex: true,
    };
    const mockProp: any = jest.fn();
    const routeProps = {
        history: mockProp,
        location: mockProp,
        match: mockProp,
        staticContext: mockProp,
    };

    it('renders', () => {
        const sideMenu = shallow(
            <StaticRouter location="/" context={{}}>
                <SideMenu
                    isCurrent={true}
                    isPurchased={true}
                    themeDesignSections={['cat', 'named', 'moe']}
                    settings={mockSettings}
                    themeId="1234567-1223-011123-111111"
                    themeName="Caterstone"
                    configurationId="4444-55555-77777-00000"
                    variationId="2222-11111-0000-44444"
                    versionId="2221-211111-0111-41111"
                    postThemeConfigData={mockpostThemeConfig}
                    {...routeProps}
                />
            </StaticRouter>
        );

        expect(sideMenu).toMatchSnapshot();
    });

    describe('Collapse Menu Button', () => {
        const sideMenu = shallow(
            <SideMenu
                isCurrent={true}
                isPurchased={true}
                themeDesignSections={['cat', 'named', 'joe']}
                settings={mockSettings}
                themeId="7777-9999-1111-0000"
                themeName="Catstone Box"
                configurationId="3333-2222-111111-00000"
                variationId="2222-11111-0000-44444"
                versionId="2221-211111-0111-41111"
                postThemeConfigData={mockpostThemeConfig}
                {...routeProps}
            />
        );

        describe('when the page is first loaded', () => {
            it('should have a state of initial', () => {
                sideMenu.setState({ collapsed: Collapsed.Initial });
                expect(sideMenu.state().collapsed).toEqual(Collapsed.Initial);
            });
        });

        describe('when the user first clicks the collapse button', () => {
            it('should set collapsed state to yes', () => {
                const collapseBtn = sideMenu.find(CollapseButton);

                sideMenu.setState({ collapsed: Collapsed.Initial });
                collapseBtn.simulate('click');
                expect(sideMenu.state().collapsed).toEqual(Collapsed.Yes);
            });
        });

        describe('when the user clicks the collapse button again to re-open menu', () => {
            it('should set collapsed state to no', () => {
                const collapseBtn = sideMenu.find(CollapseButton);

                sideMenu.setState({ collapsed: Collapsed.Initial });
                collapseBtn.simulate('click'); // collapse menu
                collapseBtn.simulate('click'); // open menu
                expect(sideMenu.state().collapsed).toEqual(Collapsed.No);
            });
        });
    });

    describe('Theme status label', () => {
        describe('when isCurrent is false and isPurchased is true', () => {
            it('renders correctly', () => {
                const sideMenu = shallow(
                    <SideMenu
                        isCurrent={false}
                        isPurchased={true}
                        themeDesignSections={['cat', 'named', 'joe']}
                        settings={mockSettings}
                        themeId="7777-9999-1111-0000"
                        themeName="Catstone Box"
                        configurationId="3333-2222-111111-00000"
                        variationId="2222-11111-0000-44444"
                        versionId="2221-211111-0111-41111"
                        postThemeConfigData={mockpostThemeConfig}
                        {...routeProps}
                    />
                );

                expect(sideMenu).toMatchSnapshot();
            });
        });

        describe('when isCurrent is true and isPurchased is true', () => {
            it('renders correctly', () => {
                const sideMenu = shallow(
                    <SideMenu
                        isCurrent={true}
                        isPurchased={true}
                        themeDesignSections={['cat', 'named', 'joe']}
                        settings={mockSettings}
                        themeId="7777-9999-1111-0000"
                        themeName="Catstone Box"
                        configurationId="3333-2222-111111-00000"
                        variationId="2222-11111-0000-44444"
                        versionId="2221-211111-0111-41111"
                        postThemeConfigData={mockpostThemeConfig}
                        {...routeProps}
                    />
                );

                expect(sideMenu).toMatchSnapshot();
            });
        });

        describe('when isPurchased is false', () => {
            it('renders correctly', () => {
                const sideMenu = shallow(
                    <SideMenu
                        isCurrent={true}
                        isPurchased={false}
                        themeDesignSections={['cat', 'named', 'joe']}
                        settings={mockSettings}
                        themeId="7777-9999-1111-0000"
                        themeName="Catstone Box"
                        configurationId="3333-2222-111111-00000"
                        variationId="2222-11111-0000-44444"
                        versionId="2221-211111-0111-41111"
                        postThemeConfigData={mockpostThemeConfig}
                        {...routeProps}
                    />
                );

                expect(sideMenu).toMatchSnapshot();
            });
        });
    });
});
