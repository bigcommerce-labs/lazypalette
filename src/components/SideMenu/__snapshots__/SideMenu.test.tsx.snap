// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu Theme status label when active theme is not the current theme and isPurchased is true renders correctly 1`] = `
<Fragment>
  <ExpandMenuRoutes
    route="styles"
  />
  <ExpandMenuRoutes
    route="history"
  />
  <ExpandMenuRoutes
    route="options"
  />
  <ExpandMenuRoutes
    key="0"
    route="section/0"
  />
  <ExpandMenuRoutes
    key="1"
    route="section/1"
  />
  <ExpandMenuRoutes
    key="2"
    route="section/2"
  />
  <styled.nav
    collapsed="initial"
  >
    <styled.div>
      <styled.div
        innerRef={
          Object {
            "current": null,
          }
        }
        theme={
          Object {
            "typography": Object {
              "fontSize": Object {
                "larger": "20px",
              },
            },
          }
        }
      >
        Catstone Box
      </styled.div>
      <Tooltip
        clickable={true}
        message="This theme is not currently applied to your store."
      >
        <styled.div
          status="INACTIVE THEME"
          theme={
            Object {
              "colors": Object {
                "empty": "#FFFFFF",
                "guideText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "success": "#FFFFFF",
              },
              "typography": Object {
                "fontSize": Object {
                  "smallest": "12px",
                },
                "fontWeight": Object {
                  "bold": "600",
                },
              },
            }
          }
        >
          INACTIVE THEME
        </styled.div>
      </Tooltip>
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "background": "#FFFFFF",
          },
        }
      }
    >
      <Route
        path="/"
        render={[Function]}
      />
    </styled.div>
    <styled.div>
      <styled.a
        href="/"
        target="_blank"
        theme={
          Object {
            "colors": Object {
              "primaryText": "#FFFFFF",
            },
            "typography": Object {
              "fontWeight": Object {
                "bold": "600",
              },
            },
          }
        }
      >
        <styled.span>
          View live store
        </styled.span>
        <styled.div
          theme={
            Object {
              "colors": Object {
                "guideText": "#FFFFFF",
              },
            }
          }
        >
          <Icon
            glyph="externalLink"
            size="small"
          />
        </styled.div>
      </styled.a>
      <Tooltip
        hideDelay={0}
        message="Collapse menu"
      >
        <styled.button
          collapsed="initial"
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "brandPrimary": "#FFFFFF",
                "empty": "#FFFFFF",
                "primaryText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "selectedBackground": "#FFFFFF",
                "stroke": "#FFFFFF",
              },
              "elevation": Object {
                "raised": "0 1px 6px rgba(48, 53, 64, 0.2)",
              },
              "typography": Object {
                "fontWeight": Object {
                  "thin": "600",
                },
              },
            }
          }
        />
      </Tooltip>
    </styled.div>
  </styled.nav>
</Fragment>
`;

exports[`SideMenu Theme status label when activeTheme is the current theme and isPurchased is true renders correctly 1`] = `
<Fragment>
  <ExpandMenuRoutes
    route="styles"
  />
  <ExpandMenuRoutes
    route="history"
  />
  <ExpandMenuRoutes
    route="options"
  />
  <ExpandMenuRoutes
    key="0"
    route="section/0"
  />
  <ExpandMenuRoutes
    key="1"
    route="section/1"
  />
  <ExpandMenuRoutes
    key="2"
    route="section/2"
  />
  <styled.nav
    collapsed="initial"
  >
    <styled.div>
      <styled.div
        innerRef={
          Object {
            "current": null,
          }
        }
        theme={
          Object {
            "typography": Object {
              "fontSize": Object {
                "larger": "20px",
              },
            },
          }
        }
      >
        Catstone Box
      </styled.div>
      <Tooltip
        clickable={true}
        message="This theme is currently applied to your store."
      >
        <styled.div
          status="ACTIVE THEME"
          theme={
            Object {
              "colors": Object {
                "empty": "#FFFFFF",
                "guideText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "success": "#FFFFFF",
              },
              "typography": Object {
                "fontSize": Object {
                  "smallest": "12px",
                },
                "fontWeight": Object {
                  "bold": "600",
                },
              },
            }
          }
        >
          ACTIVE THEME
        </styled.div>
      </Tooltip>
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "background": "#FFFFFF",
          },
        }
      }
    >
      <Route
        path="/"
        render={[Function]}
      />
    </styled.div>
    <styled.div>
      <styled.a
        href="/"
        target="_blank"
        theme={
          Object {
            "colors": Object {
              "primaryText": "#FFFFFF",
            },
            "typography": Object {
              "fontWeight": Object {
                "bold": "600",
              },
            },
          }
        }
      >
        <styled.span>
          View live store
        </styled.span>
        <styled.div
          theme={
            Object {
              "colors": Object {
                "guideText": "#FFFFFF",
              },
            }
          }
        >
          <Icon
            glyph="externalLink"
            size="small"
          />
        </styled.div>
      </styled.a>
      <Tooltip
        hideDelay={0}
        message="Collapse menu"
      >
        <styled.button
          collapsed="initial"
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "brandPrimary": "#FFFFFF",
                "empty": "#FFFFFF",
                "primaryText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "selectedBackground": "#FFFFFF",
                "stroke": "#FFFFFF",
              },
              "elevation": Object {
                "raised": "0 1px 6px rgba(48, 53, 64, 0.2)",
              },
              "typography": Object {
                "fontWeight": Object {
                  "thin": "600",
                },
              },
            }
          }
        />
      </Tooltip>
    </styled.div>
  </styled.nav>
</Fragment>
`;

exports[`SideMenu Theme status label when isPurchased is false renders correctly 1`] = `
<Fragment>
  <ExpandMenuRoutes
    route="styles"
  />
  <ExpandMenuRoutes
    route="history"
  />
  <ExpandMenuRoutes
    route="options"
  />
  <ExpandMenuRoutes
    key="0"
    route="section/0"
  />
  <ExpandMenuRoutes
    key="1"
    route="section/1"
  />
  <ExpandMenuRoutes
    key="2"
    route="section/2"
  />
  <styled.nav
    collapsed="initial"
  >
    <styled.div>
      <styled.div
        innerRef={
          Object {
            "current": null,
          }
        }
        theme={
          Object {
            "typography": Object {
              "fontSize": Object {
                "larger": "20px",
              },
            },
          }
        }
      >
        Catstone Box
      </styled.div>
      <Tooltip
        clickable={true}
        message="To use this theme, you must buy it before you can save or publish to your store."
      >
        <styled.div
          status="THEME PREVIEW"
          theme={
            Object {
              "colors": Object {
                "empty": "#FFFFFF",
                "guideText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "success": "#FFFFFF",
              },
              "typography": Object {
                "fontSize": Object {
                  "smallest": "12px",
                },
                "fontWeight": Object {
                  "bold": "600",
                },
              },
            }
          }
        >
          THEME PREVIEW
        </styled.div>
      </Tooltip>
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "background": "#FFFFFF",
          },
        }
      }
    >
      <Route
        path="/"
        render={[Function]}
      />
    </styled.div>
    <styled.div>
      <styled.a
        href="/"
        target="_blank"
        theme={
          Object {
            "colors": Object {
              "primaryText": "#FFFFFF",
            },
            "typography": Object {
              "fontWeight": Object {
                "bold": "600",
              },
            },
          }
        }
      >
        <styled.span>
          View live store
        </styled.span>
        <styled.div
          theme={
            Object {
              "colors": Object {
                "guideText": "#FFFFFF",
              },
            }
          }
        >
          <Icon
            glyph="externalLink"
            size="small"
          />
        </styled.div>
      </styled.a>
      <Tooltip
        hideDelay={0}
        message="Collapse menu"
      >
        <styled.button
          collapsed="initial"
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "brandPrimary": "#FFFFFF",
                "empty": "#FFFFFF",
                "primaryText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "selectedBackground": "#FFFFFF",
                "stroke": "#FFFFFF",
              },
              "elevation": Object {
                "raised": "0 1px 6px rgba(48, 53, 64, 0.2)",
              },
              "typography": Object {
                "fontWeight": Object {
                  "thin": "600",
                },
              },
            }
          }
        />
      </Tooltip>
    </styled.div>
  </styled.nav>
</Fragment>
`;

exports[`SideMenu renders 1`] = `
<Fragment>
  <ExpandMenuRoutes
    route="styles"
  />
  <ExpandMenuRoutes
    route="history"
  />
  <ExpandMenuRoutes
    route="options"
  />
  <ExpandMenuRoutes
    key="0"
    route="section/0"
  />
  <ExpandMenuRoutes
    key="1"
    route="section/1"
  />
  <ExpandMenuRoutes
    key="2"
    route="section/2"
  />
  <styled.nav
    collapsed="initial"
  >
    <styled.div>
      <styled.div
        innerRef={
          Object {
            "current": null,
          }
        }
        theme={
          Object {
            "typography": Object {
              "fontSize": Object {
                "larger": "20px",
              },
            },
          }
        }
      >
        Caterstone
      </styled.div>
      <Tooltip
        clickable={true}
        message="This theme is currently applied to your store."
      >
        <styled.div
          status="ACTIVE THEME"
          theme={
            Object {
              "colors": Object {
                "empty": "#FFFFFF",
                "guideText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "success": "#FFFFFF",
              },
              "typography": Object {
                "fontSize": Object {
                  "smallest": "12px",
                },
                "fontWeight": Object {
                  "bold": "600",
                },
              },
            }
          }
        >
          ACTIVE THEME
        </styled.div>
      </Tooltip>
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "background": "#FFFFFF",
          },
        }
      }
    >
      <Route
        path="/"
        render={[Function]}
      />
    </styled.div>
    <styled.div>
      <styled.a
        href="/"
        target="_blank"
        theme={
          Object {
            "colors": Object {
              "primaryText": "#FFFFFF",
            },
            "typography": Object {
              "fontWeight": Object {
                "bold": "600",
              },
            },
          }
        }
      >
        <styled.span>
          View live store
        </styled.span>
        <styled.div
          theme={
            Object {
              "colors": Object {
                "guideText": "#FFFFFF",
              },
            }
          }
        >
          <Icon
            glyph="externalLink"
            size="small"
          />
        </styled.div>
      </styled.a>
      <Tooltip
        hideDelay={0}
        message="Collapse menu"
      >
        <styled.button
          collapsed="initial"
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "brandPrimary": "#FFFFFF",
                "empty": "#FFFFFF",
                "primaryText": "#FFFFFF",
                "secondaryText": "#FFFFFF",
                "selectedBackground": "#FFFFFF",
                "stroke": "#FFFFFF",
              },
              "elevation": Object {
                "raised": "0 1px 6px rgba(48, 53, 64, 0.2)",
              },
              "typography": Object {
                "fontWeight": Object {
                  "thin": "600",
                },
              },
            }
          }
        />
      </Tooltip>
    </styled.div>
  </styled.nav>
</Fragment>
`;
